ClangTidy warnings for `cppcoreguidelines-avoid-non-const-global-variables`

```
/Users/jviotti/Projects/core/src/core/json/include/sourcemeta/core/json.h:48:6: error: variable 'parse_json' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   48 | auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
      |      ^

/Users/jviotti/Projects/core/src/core/json/include/sourcemeta/core/json.h:88:6: error: variable 'parse_json' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   88 | auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
      |      ^

/Users/jviotti/Projects/core/src/core/json/json.cc:18:6: error: variable 'parse_json' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   18 | auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
      |      ^

/Users/jviotti/Projects/core/src/core/json/json.cc:30:6: error: variable 'parse_json' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   30 | auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
      |      ^

/Users/jviotti/Projects/core/src/core/json/parser.h:673:6: error: variable 'internal_parse_json' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
  673 | auto internal_parse_json(
      |      ^

/Users/jviotti/Projects/core/src/core/json/stringify.h:23:6: error: variable 'indent' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   23 | auto indent(std::basic_ostream<CharT, Traits> &stream,
      |      ^

/Users/jviotti/Projects/core/src/core/jsonpointer/parser.h:20:1: error: variable 'reset' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   20 | reset(std::basic_stringstream<CharT, Traits, Allocator<CharT>> &stream)
      | ^

/Users/jviotti/Projects/core/src/core/jsonpointer/parser.h:29:1: error: variable 'parse_index' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   29 | parse_index(std::basic_stringstream<CharT, Traits, Allocator<CharT>> &stream,
      | ^

/Users/jviotti/Projects/core/src/core/jsonpointer/parser.h:44:6: error: variable 'parse_pointer' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   44 | auto parse_pointer(std::basic_istream<JSON::Char, JSON::CharTraits> &stream)
      |      ^

/Users/jviotti/Projects/core/src/core/jsonpointer/stringify.h:18:1: error: variable 'write_character' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   18 | write_character(std::basic_ostream<JSON::Char, JSON::CharTraits> &stream,
      | ^

/Users/jviotti/Projects/core/src/core/yaml/include/sourcemeta/core/yaml.h:42:6: error: variable 'parse_yaml' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
   42 | auto parse_yaml(std::basic_istream<JSON::Char, JSON::CharTraits> &stream)
      |      ^

/Users/jviotti/Projects/core/src/core/yaml/yaml.cc:110:6: error: variable 'parse_yaml' is non-const and globally accessible, consider making it const [cppcoreguidelines-avoid-non-const-global-variables,-warnings-as-errors]
  110 | auto parse_yaml(std::basic_istream<JSON::Char, JSON::CharTraits> &stream)
      |      ^

```

--- a/src/core/json/json.cc
+++ b/src/core/json/json.cc
@@ -15,6 +15,7 @@
 
 namespace sourcemeta::core {
 
+//NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
                 std::uint64_t &line, std::uint64_t &column,
                 const JSON::ParseCallback &callback) -> JSON {
@@ -27,6 +28,7 @@ auto parse_json(const std::basic_string<JSON::Char, JSON::CharTraits> &input,
   return internal_parse_json(input, line, column, callback);
 }
 
+//NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
                 const JSON::ParseCallback &callback) -> JSON {
   std::uint64_t line{1};

--- a/src/core/json/json.cc
+++ b/src/core/json/json.cc
@@ -15,7 +15,7 @@
 
 namespace sourcemeta::core {
 
-//NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
                 std::uint64_t &line, std::uint64_t &column,
                 const JSON::ParseCallback &callback) -> JSON {
@@ -28,7 +28,7 @@ auto parse_json(const std::basic_string<JSON::Char, JSON::CharTraits> &input,
   return internal_parse_json(input, line, column, callback);
 }
 
-//NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 auto parse_json(std::basic_istream<JSON::Char, JSON::CharTraits> &stream,
                 const JSON::ParseCallback &callback) -> JSON {
   std::uint64_t line{1};

--- a/src/core/yaml/yaml.cc
+++ b/src/core/yaml/yaml.cc
@@ -109,6 +109,7 @@ auto internal_parse_json(yaml_parser_t *parser) -> sourcemeta::core::JSON {
 
 namespace sourcemeta::core {
 
+//NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 auto parse_yaml(std::basic_istream<JSON::Char, JSON::CharTraits> &stream)
     -> JSON {
   std::basic_ostringstream<JSON::Char, JSON::CharTraits> buffer;

--- a/src/core/yaml/yaml.cc
+++ b/src/core/yaml/yaml.cc
@@ -109,7 +109,7 @@ auto internal_parse_json(yaml_parser_t *parser) -> sourcemeta::core::JSON {
 
 namespace sourcemeta::core {
 
-//NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
+// NOLINTNEXTLINE(cppcoreguidelines-avoid-non-const-global-variables)
 auto parse_yaml(std::basic_istream<JSON::Char, JSON::CharTraits> &stream)
     -> JSON {
   std::basic_ostringstream<JSON::Char, JSON::CharTraits> buffer;

